jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryClient.java

```diff
@@ -45,6 +45,29 @@
 
   private static final String PROTOCOL = "https";
 
+  @Nullable private static String userAgentSuffix = "";
+
+  // TODO: Inject via a RegistryClientFactory.
+  /** Sets a suffix to append to {@code User-Agent} headers. */
+  public static void setUserAgentSuffix(@Nullable String userAgentSuffix) {
+    RegistryClient.userAgentSuffix = userAgentSuffix;
+  }
+
+  // TODO: Add more descriptiveness.
```

Comment:
What do you mean? The user-agent shouldn't be too long.

Reply:
Not sure yet, removing this for now